# Завдання №1 

## Завдання 1 " Конструкції Python"


In [2]:
x = 22
y = 6.88
s = "Ігор"
a = [x, y, s]
b = True
t = (y, "Студент")

print("Тип змінної x:", type(x))
print("Тип змінної y:", type(y))
print("Тип змінної s:", type(s))
print("Тип змінної a:", type(a))
print("Тип змінної b:", type(b))
print("Тип змінної t:", type(t))

Тип змінної x: <class 'int'>
Тип змінної y: <class 'float'>
Тип змінної s: <class 'str'>
Тип змінної a: <class 'list'>
Тип змінної b: <class 'bool'>
Тип змінної t: <class 'tuple'>


In [4]:
print(x+y)
print(x-y)
print(x/y)
print(x%y)
print(x*y)
print(x//y)
print(x**y)

28.88
15.120000000000001
3.197674418604651
1.3600000000000003
151.35999999999999
3.0
1721345482.2757888


In [5]:
print(x==y)
print(x!=y)
print(x>y)
print(x<y)
print(x<=y)
print(x>=y)

False
True
True
False
False
True


In [7]:
print(str(x))
print(str(t))
print(int(x))
#print(int(s)) неможливо перетворити str в int 
print(float(x))
print(float(y))

22
(6.88, 'Студент')
22
22.0
6.88


## Завдання 3 "Введення/вивід з консолі"

In [3]:
e = float(input())


2.71


In [4]:
print(e**5)

146.1660310351


In [ ]:
name = "Ігор"
surname = "Бердник"
print (name, surname)
print("1+1", 2, sep=" -> ")


У функції print() символ , розділяє аргументи функції, які відображаються в окремих рядках, а аргумент sep визначає роздільник між ними.

In [10]:
print("{} {} - це студент".format(name, surname))
print("Результат буде = {}".format(42 / 5 + 4))
b = True
print("{} - {} хлопець!".format(name, b))
print("{} {}".format(surname, name))

Ігор Бердник - це студент
Результат буде = 12.4
Ігор - True хлопець!
Бердник Ігор


## Завдання 4 "Керування обчисленнями"


In [2]:
x = float(input("Введи x: "))

# ----- 1 ----------
if x > 5:
    print (x)
    
# ----- 2 ----------    
if -5 < x < 5 :
    print (x)
else:
    print (x ** 2)
    
# ----- 3 ----------
if x > -5:
    print (x)
elif -5 < x < 5:
    print (x ** 2)
elif x > 5:
    print (x ** (1/2))

Введи x: 31.5
31.5
992.25
31.5


x = -20 Спрацював лише другий if з умовою else

x = 31.5 Спрацювали всі if, а саме: 1-if, 2-if друга умова, 3-if перша умова

x = 4 Спрацювали 2 if перша умова та 3 if перша умова

In [7]:
age = int(input("Твій вік "))
if 0 <= age <=6:
    print('малятко')
elif 7<= age <=17:
    print('школяр')
else:
    print('Вік живи, вік учись, ти студент')


Твій вік 99
Вік живи, вік учись, ти студент


In [6]:
for i in range(10):
    if i % 2 != 0:
        print (-i)
    else:
        pass

-1
-3
-5
-7
-9


Даний код знаходиться всередині циклу for, який виконується 10 разів, кожен раз створюється змінна і. Потім перевіряєтсья умова: якшо число не парне - прінтується це число з мінусом.

In [9]:
sum = 0
for i in range(10, 101):
    if i % 3 == 0:
        sum += i
print(sum)

1665


In [10]:
while(input("тисни тут: ") != '0'):
    print ("було введено")

тисни тут: 5
було введено
тисни тут: 0


Цей код запускає нескінченний цикл, доки користувач не введе '0'

In [11]:
i = 10
total = 0
while i <= 100:
    if i % 3 == 0:
        total += i
    i += 1
print(total)

1665


# Завдання №2


## Завдання 1 "Операції над списками"
 

In [1]:
i_am = []
i_am.append('Ігор')
print(id(i_am))

2222403196032


In [5]:
i_am.append('Бердник')
print(i_am, id(i_am))

['Ігор', 'Бердник', 'Бердник', 'Бердник', 'Бердник'] 2222403196032


## Завданяя 2 "Срізи в списках" 


In [3]:
my_list = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
print(my_list[:5])
print(my_list[-6:])
for num in my_list:
    if num % 2 == 0:
        print(num)

[1, 2, 3, 4, 5]
[5, 6, 7, 8, 9, 10]
2
4
6
8
10


In [7]:
my_list = [[1, 2, 3], [4, 5, 6], [7, 8, 9]]
my_list[1][0] = 42
print(my_list)

[[1, 2, 3], [42, 5, 6], [7, 8, 9]]


In [8]:
my_list[0][2], my_list[1][1] = my_list[1][1], my_list[0][2]
print(my_list)

[[1, 2, 5], [42, 3, 6], [7, 8, 9]]


# Завдання №3 "Операції з файлами"


## Завдання 1 "Запис в файл" 

In [14]:
file = open('my_file.txt', 'w+')
for i in range(1, 11):
    file.write(f"{i}-й рядок\n")
file.close()

## Завдання 2 "Читання з файла"  

In [15]:
file = open('my_file.txt', 'r')
for i in range(5):
        print(file.readline().strip())
file.close()

1-й рядок
2-й рядок
3-й рядок
4-й рядок
5-й рядок


## Завдання 2 "Контекстні менеджери"  

In [ ]:
with open('my_file.txt', 'r') as file:
    for i in range(5):
        print(file.readline().strip())